SampleApplications/SampleLibraries/Client/Session.cs

```diff
@@ -1964,7 +1964,10 @@ public ReferenceDescriptionCollection FetchReferences(NodeId nodeId)
             bool requireEncryption = securityPolicyUri != SecurityPolicies.None;
             if (!requireEncryption)
             {
-                requireEncryption = identityPolicy.SecurityPolicyUri != SecurityPolicies.None;
+                // OPC-UA 1.03 Part 4 7.37: null or empty string => fall back to SecureChannel policy
+                requireEncryption = identityPolicy.SecurityPolicyUri != null &&
```

Comment:
Maybe you should use [String.IsNullOrEmpty](https://msdn.microsoft.com/de-de/library/system.string.isnullorempty(v=vs.110).aspx) of [String.IsNullOrWhiteSpace](https://msdn.microsoft.com/de-de/library/system.string.isnullorwhitespace(v=vs.110).aspx) instead of the check for null and then checking for an empty string